src/lava/magma/core/model/py/ports.py

```diff
@@ -108,50 +241,89 @@ def peek(self) -> ty.Tuple[int, int]:
 PyInPort.SCALAR_SPARSE = PyInPortScalarSparse
 
 
-class PyOutPort(AbstractPyPort):
-    """Python implementation of OutPort used within AbstractPyProcessModels."""
+class PyOutPort(AbstractPyIOPort):
+    """Python implementation of OutPort used within AbstractPyProcessModels.
+
+    PyOutPort is an output Port processing data sent from a connected input Port
+    (PyInPort) over a channel. PyOutPort can send (send()) the data, which
+    adds it to the channel or clear (flush()) the channel to remove any data of
+    the channel.
+    The different class attributes are used to select the type of OutPorts via
+    LavaPyType declarations in PyProcModels, e.g., LavaPyType(
+    PyOutPort.VEC_DENSE, np.int32, precision=24) creates a PyOutPort.
+    A PyOutPort (source) can be connected to one or multiple PyInPorts (target).
+
```

Comment:
"PyOutPort is an output Port processing data sent from a connected input Port (PyInPort) over a channel."
I thought that an OutPort sends data, and does not receive data from an InPort?

"PyOutPort can send (send()) the data **by adding it to the channel**, or **it can** clear (flush()) the channel to remove any data **from it.**"